### PR TITLE
crowcpp-crow: update dependencies, fix requires names

### PR DESCRIPTION
**Pull request by toge**
Specify library name and version:  **crowcpp-crow/***

---

- [x] I've read the [contributing guidelines](https://github.com/conan-io/conan-center-index/blob/master/CONTRIBUTING.md).
- [x] I've used a [recent](https://github.com/conan-io/conan/releases/latest) Conan client version close to the [currently deployed](https://github.com/conan-io/conan-center-index/blob/master/.c3i/config_v1.yml#L6).
- [x] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/crowcpp-crow/all/conanfile.py [hunk 1]
```diff
@@ -42,12 +42,12 @@ def layout(self):
...
     def requirements(self):
-        self.requires("boost/1.81.0")
+        self.requires("boost/1.83.0")
         if self.version == "0.2":
-            self.requires("openssl/1.1.1s")
+            self.requires("openssl/[>=1.1 <4]")
         if Version(self.version) >= "1.0":
             if self.options.with_ssl:
-                self.requires("openssl/1.1.1s")
+                self.requires("openssl/1.1.1v")
```
Comment (RubenRBS) [inline recipes/crowcpp-crow/all/conanfile.py:50]: Why does 0.2 allow for version range but no >=1.0?

**Review by RubenRBS — COMMENTED** [review]
Thanks! Question about the openssl range :)

File: recipes/crowcpp-crow/all/conanfile.py (hunk 1 above)
Comment (toge) [reply recipes/crowcpp-crow/all/conanfile.py:50]: @RubenRBS 
I met following execution error with OpenSSL/3.x.x:
```
terminate called after throwing an instance of 'boost::wrapexcept<boost::system::system_error>'
  what():  context: unknown option (digital envelope routines) [asio.ssl:50331817]
```

It seems to be caused by OpenSSL/3.x.x providers.
https://wiki.openssl.org/index.php/OpenSSL_3.0#Providers

Comment (uilianries) [reply recipes/crowcpp-crow/all/conanfile.py:50]: @toge you still can use ranges, but in that case: `openssl/[>=1.1 <3]`

Comment (uilianries) [inline recipes/crowcpp-crow/all/conanfile.py:50]:
```suggestion
                self.requires("openssl/[>=1.1 <3]")
```

Comment (toge) [reply recipes/crowcpp-crow/all/conanfile.py]: @uilianries 
Thanks a lot!

Comment (RubenRBS) [reply recipes/crowcpp-crow/all/conanfile.py]: Weird that an old version supports openssl 3, but newer ones dont!

**Review by RubenRBS — APPROVED** [review]

**Pull request merged**